### PR TITLE
[Elephant] Issue #193 Updated AccountPropertyTransaction Names

### PR DESCRIPTION
**Pull request by rg911**
Issue #193

Updated `AccountPropertyTransaction` name to `AccountRestrictionTransaction

- Applied new DTOs from nem2-doc
- Updated all account property models
- Updated account property transaction builders, schemas and buffers
- Updated all comments texts
- Updated all tests

### CODE REVIEW

**Comment by dgarcia360** [comment]
Note that ``EntityType`` has been renamed to ``Operation``: https://github.com/nemtech/catbuffer/blob/master/schemas/restriction_account/account_operation_restriction.cats

File: e2e/infrastructure/AccountHttp.spec.ts [hunk 1]
```diff
@@ -286,7 +286,7 @@ describe('AccountHttp', () => {
         it('should call getAccountProperty successfully', (done) => {
```
Comment (evias) [inline e2e/infrastructure/AccountHttp.spec.ts]: rename of getAccountProperty not complete ?

File: e2e/infrastructure/AccountHttp.spec.ts [hunk 2]
```diff
@@ -286,7 +286,7 @@ describe('AccountHttp', () => {
...
             setTimeout(() => {
                 accountHttp.getAccountProperties(accountAddress).subscribe((accountProperty) => {
```
Comment (evias) [inline e2e/infrastructure/AccountHttp.spec.ts]: same here

File: src/infrastructure/AccountHttp.ts [hunk 3]
```diff
@@ -97,7 +97,7 @@ export class AccountHttp extends Http implements AccountRepository {
      * @param publicAccount public account
      * @returns Observable<AccountProperty>
      */
-    public getAccountProperties(address: Address): Observable<AccountPropertiesInfo> {
+    public getAccountProperties(address: Address): Observable<AccountRestrictionsInfo> {
```
Comment (evias) [inline src/infrastructure/AccountHttp.ts]: rename to `getAccountRestrictions` not done in swagger.yaml ?

File: src/model/account/RestrictionType.ts [hunk 4]
```diff
@@ -15,15 +15,15 @@
...
+ * 0x03	The restriction type is a transaction type.
+ * 0x04	restriction type sentinel.
+ * 0x80 + type	The restriction is interpreted as a blocking operation.
  */
 
-export enum PropertyType {
+export enum RestrictionType {
     AllowAddress = 0x01,
     AllowMosaic = 0x02,
     AllowTransaction = 0x04,
```
Comment (evias) [inline src/model/account/RestrictionType.ts]: consistency problem: in doccomment `0x04` is "restriction type sentinel" while here it represents a transaction type.

File: test/model/transaction/AccountPropertyTransaction.spec.ts [hunk 5]
```diff
@@ -17,95 +17,95 @@
...
+        it('should return 148 for AccountAddressRestrictionModificationTransaction transaction byte size with 1 modification', () => {
             const address = Address.createFromRawAddress('SBILTA367K2LX2FEXG5TFWAS7GEFYAGY7QLFBYKC');
-            const addressPropertyFilter = AccountPropertyModification.createForAddress(
-                PropertyModificationType.Add,
+            const addressPropertyFilter = AccountRestrictionModification.createForAddress(
+                RestrictionModificationType.Add,
                 address,
             );
-            const addressPropertyTransaction = AccountPropertyTransaction.createAddressPropertyModificationTransaction(
+            const addressPropertyTransaction = AccountRestrictionTransaction.createAddressPropertyModificationTransaction(
```
Comment (evias) [inline test/model/transaction/AccountPropertyTransaction.spec.ts]: rename to `createAddressRestrictionModificationTransaction`

File: test/model/transaction/AccountPropertyTransaction.spec.ts [hunk 6]
```diff
@@ -17,95 +17,95 @@
...
+        it('should return 131 for AccountMosaicRestrictionModificationTransaction transaction byte size with 1 modification', () => {
             const mosaicId = new MosaicId([2262289484, 3405110546]);
-            const mosaicPropertyFilter = AccountPropertyModification.createForMosaic(
-                PropertyModificationType.Add,
+            const mosaicPropertyFilter = AccountRestrictionModification.createForMosaic(
+                RestrictionModificationType.Add,
                 mosaicId,
             );
-            const mosaicPropertyTransaction = AccountPropertyTransaction.createMosaicPropertyModificationTransaction(
+            const mosaicPropertyTransaction = AccountRestrictionTransaction.createMosaicPropertyModificationTransaction(
```
Comment (evias) [inline test/model/transaction/AccountPropertyTransaction.spec.ts]: rename to `createMosaicRestrictionModificationTransaction`

File: test/model/transaction/AccountPropertyTransaction.spec.ts [hunk 7]
```diff
@@ -235,13 +235,13 @@ describe('AccountPropertyTransaction', () => {
...
 
         const entityType = TransactionType.ADDRESS_ALIAS;
-        const entityTypePropertyFilter = AccountPropertyModification.createForEntityType(
-            PropertyModificationType.Add,
+        const entityTypePropertyFilter = AccountRestrictionModification.createForEntityType(
+            RestrictionModificationType.Add,
             entityType,
         );
-        const entityTypePropertyTransaction = AccountPropertyTransaction.createEntityTypePropertyModificationTransaction(
+        const entityTypePropertyTransaction = AccountRestrictionTransaction.createEntityTypePropertyModificationTransaction(
```
Comment (evias) [inline test/model/transaction/AccountPropertyTransaction.spec.ts]: rename with `Restriction` in function name

**Pull request merged**